### ISSUE TITLE
Feature: add a way to customize basic styling

### ISSUE THREAD

**Issue by ChopinDavid** [goxiaoy/flutter_survey_js#14]
Right now, the `Text` widgets that display each survey element have hard-coded styling. For example, the `QuestionTitle` widget uses the following lines of code to style its `Text`:

```dart
final titleTextStyle = () {
  return TextStyle(
    fontSize: 16.0,
    fontFamily: 'SF-UI-Text',
    fontWeight: FontWeight.w900,
    color: Theme.of(context).disabledColor //Color(0xff242833)
  );
};

...

//later in the code:
Text(
  '${status.indexAll! + 1}.',
  style: titleTextStyle(),
)
```

I am envisioning a class where we could define the styles for basic UI elements, such as question titles. Implementation would look something like this:
```dart
class SurveyStylesConfiguration {
  SurveyStylesConfiguration({this.questionTitleTextStyle});
  TextStyle? questionTitleTextStyle;
}

class SomeWidget extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return  SurveyWidget(
      survey: survey!,
      stylesConfiguration:
        SurveyStylesConfiguration(
          questionTitleTextStyle: TextStyle(
            fontSize: 24.0,
            fontFamily: 'Helvetica',
            fontWeight: FontWeight.w700,
            color: Colors.grey,
          ),
    );
}
```

**Comment by ChopinDavid** [comment]
Ideally, there is a way for the Text widgets within survey elements to match the app's [`Theme`](https://docs.flutter.dev/cookbook/design/themes).

**Comment by goxiaoy** [comment]
Another way to customize is providing params like `titleBuilder`  in the constructor
```dart
typedef Widget TitleBuilder(
  BuildContext context,
  String title,
  ...
);
```

Use default builder behavior if not set

**Comment by laogao** [comment]
> Ideally, there is a way for the `Text` widgets within survey elements to match the app's [`Theme`](https://docs.flutter.dev/cookbook/design/themes).

I'd like to discuss more on this topic.

Lately we've rec'd a user feedback that the question titles were somewhat weak and not obvious. The problem is with the use of `Theme.of(context).disabledColor` which I believe is indented for "disabled" controls. Hence more often than not we're getting something "greyish". We managed to circumvent this by wrapping the `SurveyWidget` inside a copied `Theme` with `disabledColor` replaced by a more "heavy", "prominent" color. But this whole process felt tedious and at the same time underwhelming.

If we aim to match the app's `Theme`, which I think would be ideal, maybe the first thing we could do is pick a more "standard" naming/theming mechanism, e.g. `Theme.of(context).textTheme.titleLarge` for question titles? Once we have all UI elements follow the theme inherited from context, we can let the user tweak the styling from the outside rather than effectively build the whole thing by providing each and every builder there is.

Don't get me wrong. Builders are powerful tools. But sometimes user only needs to change a color, so to speak. Thoughts?

**Comment by ChopinDavid** [comment]
@laogao I agree, `Theme.of(context).disabledColor` feels like the incorrect `Color` for these `TextStyle`s. This is probably an artifact of how goxiaoy wanted titles to look for his specific use case but has not been adapted for more general use. This issue should hopefully resolve that.

Our team has gotten around the issue by removing `color: Theme.of(context).disabledColor` entirely from any occurrence in a `TextStyle` constructor, [see here](https://github.com/wwt/flutter_survey_js/commit/d2087a08723e0298a6691d4edfa0be865144f04a).

```dart
final titleTextStyle = () {
  return TextStyle(
    fontSize: 16.0,
    fontFamily: 'SF-UI-Text',
    fontWeight: FontWeight.w900,
    color: Theme.of(context).disabledColor //Color(0xff242833)
  );
};
```
turns into
```dart
final textStyle = Theme.of(context).textTheme.bodyText1;
```

and all instances of `titleTextStyle()` turn into `textStyle`.

With [this subsequent commit](https://github.com/wwt/flutter_survey_js/commit/105a3b4adea57dfb605ad5f25f38dac3bfd0c607), we ultimately `textStyle` back to `titleTextStyle()`, etc., but are now using `Theme.of(context).textTheme.headlineMedium` to create `titleTextStyle()`.

With our current implementation, we can do something like:
```dart
MaterialApp(
  ...
  theme: ThemeData(
    textTheme: TextTheme(
      headlineMedium: TextStyle(
        color: Colors.red,
      ),
    ),
  ),
  ...
)
```
to give all of our titles a red color:
<img width="420" alt="Screen Shot 2023-05-05 at 1 09 24 PM" src="https://user-images.githubusercontent.com/27712906/236535350-bbea4f24-1509-4fae-b5c4-6612e89eb438.png">

Using the app theme to set element title colors, element description colors, etc. does have the one draw back of making all other `Text` widgets that use the same `TextStyle` update, i.e. if we use `Theme.of(context).headlineMedium` to make all of our titles red, every other, possibly unrelated, `Text` widget using `Theme.of(context).headlineMedium` as its `textStyle` will also turn red.

One workaround for this was to create something like a `SurveyStylesConfiguration` that can be provided at the top level that allows us to specify our own `TextStyle`s, such as `questionTitleTextStyle`, without tampering with the more broadly used `TextStyle`s that one would grab off of `Theme.of(context).textTheme`.

**Comment by laogao** [comment]
@ChopinDavid Initially I also had concerns about messing with app global theme settings, until I realized the theme can have narrower scopes by wrapping the part you want to be affected by your customization in its own parent `Theme` widget, and tweak however you like from there, without accidentally change unrelated parts of your app. See https://docs.flutter.dev/cookbook/design/themes#themes-for-part-of-an-application .

**Comment by goxiaoy** [comment]
@laogao question title use  `Theme.of(context).textTheme.titleMedium` and description use `Theme.of(context).textTheme.bodyMedium` temporally

Maybe we can add a `TitleBuilder`  configuration later

**Comment by laogao** [comment]
> @laogao question title use `Theme.of(context).textTheme.titleMedium` and description use `Theme.of(context).textTheme.bodyMedium` temporally
> 
> Maybe we can add a `TitleBuilder` configuration later

@goxiaoy Appreciated for the quick response.

Overall I think things are definitely moving in the right direction. One small caveat though: `titleMedium` has already been used by quite a few form widgets that frequently show up in a survey, such as `CheckboxListTile` and `RadioListTile`, etc. Which means once I change the `titleMedium`, both the title and the choices below the title are affected (they look the same).

I'd like to propose a change to use `titleLarge` instead, which could be a better default. If one feels this title to be too large, he/she can easily tweak to his/her liking. A PR is underway for your consideration.

Cheers.
